### PR TITLE
Gi JOE sheet improvements

### PR DESCRIPTION
**Pull request by phildominguez**
Addresses https://github.com/WookieeMatt/Essence20/issues/193

Since I needed to have the Defense for non PR changes in order to do the GI Joe Character sheet.

### CODE REVIEW

File: templates/actor/parts/headers/actor-gijoe-header.hbs [hunk 1]
```diff
@@ -1 +1,24 @@
...
+<div>
+  <label for="system.languages" class="resource-label">{{localize 'E20.ActorLanguages'}}</label>
+  <input type="text" name="system.languages" value="{{system.languages}}" />
+</div>
+<div class="pr-header-level">
+  <label for="system.level" class="resource-label">{{localize 'E20.ActorLevel'}}</label>
+  <input class="centered-input" style="width: 100%;" type="number" name="system.level" value="{{system.level}}" />
+</div>
+{{/inline}}
+{{/"systems/essence20/templates/actor/parts/headers/actor-header.hbs"}}
```
Comment (phildominguez) [inline templates/actor/parts/headers/actor-gijoe-header.hbs:24]: If you hover the ⛔ you can see it's warning you about a missing newline. [Best practice](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline) is to end all files with newlines. If you're using VSCode, go to File -> Preferences -> Settings, search for "insert final newline" and enable that to automatically add them.

File: templates/actor/parts/actor-defenses.hbs [hunk 2]
```diff
@@ -21,37 +21,24 @@
...
+             {{sum defense.value @root.equippedArmorEffect}}
           </span>
         </td>
         {{else}}
         {{!-- Remaining Defenses --}}
         <td class="tg-td" style="border-color: {{system.color}};">
           <span
-            title="{{defense.unmorphed.value}} {{localize 'E20.DefenseUnmorphed'}} + {{defense.morphed.bonus}} {{localize 'E20.DefenseBonus'}}">
-            {{sum defense.morphed.value}}
+            title="{{@root.config.defenseBase}} {{localize 'E20.DefenseBase'}} + + {{lookup @root.system.essences defense.essence}} {{localize (lookup @root.config.essences defense.essence)}} + {{defense.bonus}} {{localize 'E20.DefenseBonus'}}">
```
Comment (phildominguez) [inline templates/actor/parts/actor-defenses.hbs:40]: `+ +` here
image

File: templates/actor/parts/actor-defenses.hbs [hunk 3]
```diff
@@ -21,37 +21,24 @@
 
     {{!-- Body --}}
     <tbody>
-      {{!-- Unmorphed row --}}
+        {{!-- Morphed row --}}
```
Comment (phildominguez) [inline templates/actor/parts/actor-defenses.hbs:24]: Can remove this

File: templates/actor/parts/actor-defenses.hbs [hunk 4]
```diff
@@ -21,37 +21,24 @@
...
-        <td class="tg-td" style="border-color: {{system.color}};">{{localize 'E20.DefenseMorphed'}}</td>
+        <td class="tg-td" style="border-color: {{system.color}};">{{localize 'E20.DefenseValues'}}</td>
         {{#each system.defenses as |defense|}}
         {{!-- Special case for Toughness to add armor --}}
         {{#ifEquals defense.name 'toughness'}}
         <td class="tg-td" style="border-color: {{system.color}};">
           <span
-            title="{{defense.unmorphed.value}} {{localize 'E20.DefenseUnmorphed'}} + {{defense.morphed.bonus}} {{localize 'E20.DefenseBonus'}} + {{@root.equippedArmorEffect}} {{localize 'ITEM.TypeArmor'}}">
-            {{sum defense.morphed.value @root.equippedArmorEffect}}
+            title="{{@root.config.defenseBase}} {{localize 'E20.DefenseBase'}} + {{lookup @root.system.essences defense.essence}} {{localize (lookup @root.config.essences defense.essence)}} + {{defense.morphed.bonus}} {{localize 'E20.DefenseBonus'}} + {{@root.equippedArmorEffect}} {{localize 'ITEM.TypeArmor'}}">
```
Comment (phildominguez) [inline templates/actor/parts/actor-defenses.hbs:32]: Contains `{{defense.morphed.bonus}} {{localize 'E20.DefenseBonus'}}`

File: module/helpers/templates.mjs [hunk 5]
```diff
@@ -24,6 +24,7 @@ export const preloadHandlebarsTemplates = async function () {
     "systems/essence20/templates/actor/parts/actor-collapsible-item-container.hbs",
     "systems/essence20/templates/actor/parts/actor-common.hbs",
     "systems/essence20/templates/actor/parts/actor-defenses.hbs",
+    "systems/essence20/templates/actor/parts/actor-pr-defenses.hbs",
```
Comment (phildominguez) [inline module/helpers/templates.mjs:27]: I think I prefer `actor-defenses-pr.hbs`, that way all defense-related files are next to each other alphabetically.

File: lang/en.json [hunk 6]
```diff
@@ -119,6 +121,7 @@
     "ItemSourcePage": "Page",
     "ItemToggleEquipped": "Toggle equipped",
     "ItemTypeArmorPlural": "Armor",
+    "ItemTypeBattleDress": "Battledress",
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply lang/en.json:124]: it is what GI Joe calls armor. I am still trying to decide if we should use it.

File: templates/actor/parts/headers/actor-gijoe-header.hbs (hunk 1 above)
Comment (JFranklin3000) [reply templates/actor/parts/headers/actor-gijoe-header.hbs:24]: fixed

File: templates/actor/parts/actor-defenses.hbs (hunk 2 above)
Comment (JFranklin3000) [reply templates/actor/parts/actor-defenses.hbs:40]: fixed

File: templates/actor/parts/actor-defenses.hbs (hunk 4 above)
Comment (JFranklin3000) [reply templates/actor/parts/actor-defenses.hbs:32]: fixed

Comment (phildominguez) [reply templates/actor/parts/actor-defenses.hbs:32]: Still seeing `defense.morphed.bonus`

Comment (phildominguez) [reply templates/actor/parts/actor-defenses.hbs:32]: image

Comment (JFranklin3000) [reply templates/actor/parts/actor-defenses.hbs]: should be right now I must have fixed in the other branch.

**Comment by phildominguez** [comment]
Approved!

**Pull request merged**